lib/build.js

```diff
@@ -1,208 +1,133 @@
-                    container,
-                    jobId,
-                    jobName: response.jobName,
-                    pipelineId: response.pipelineId,
-                    scmUrl: response.scmUrl
-                }])
-                .then(() => nodeify.success(response))
-            )
-            .then((response) =>
-                this.pipeline.getAdmin(response.pipelineId)
-                    .then((admin) => nodeify.success(hoek.applyToDefaults(response, { admin })))
-            )
-            .then((response) =>
-                // Create github status
-                nodeify.withContext(githubHelper, 'run', [{
-                    user: this.user,
-                    username: response.admin,
-                    action: 'createStatus',
-                    params: {
-                        user: response.user,
-                        repo: response.repo,
-                        sha: initialBuildData.sha,
-                        state: 'pending',
-                        context: 'screwdriver'
-                    }
-                }])
-            )
-            .then(() => nodeify.success(hoek.applyToDefaults({ id }, initialBuildData), callback))
-            .catch((err) => nodeify.fail(err, callback));
+        return job;
     }
 
     /**
-     * Gets all the builds for a jobId
-     * @method getBuildsForJobId
-     * @param  {Object}   config                Config object
-     * @param  {String}   config.jobId          The jobId of the build to filter for
-     * @param  {Function} [callback]            The callback object to return the stream to
-     * @return {Promise}                        If no callback is given, a Promise is returned
+     * Lazy load a user model
+     * @property user
+     * @return {Promise}    Resolves to the user associated with this build
      */
-    getBuildsForJobId(config, callback) {
-        const listConfig = {
-            params: {
-                jobId: config.jobId
-            },
-            paginate: {
-                count: 25, // This limit is set by the matrix restriction
-                page: 1
-            }
-        };
+    get user() {
+        delete this.user;
+        const factory = new UserFactory(this.datastore);
+        const user = factory.get({ username: this.username });
+
+        // ES6 has weird getters and setters in classes,
+        // so we redefine the pipeline property here to resolve to the
+        // resulting promise and not try to recreate the factory, etc.
+        Object.defineProperty(this, 'user', {
+            enumerable: true,
+            value: user
+        });
+
+        return user;
+    }
 
-        return this.list(listConfig)
-            .then((records) => {
-                records.sort((build1, build2) => build1.number - build2.number);
+    /**
+     * Start this build
+     * @method start
+     * @return {Promise}
+     */
+    start() {
+        const userFactory = new UserFactory(this.datastore);
```

Comment:
I find it kinda awkward that we're instantiating a new userfactory a couple of times throughout this build.

Does it make sense to instantiate it as a part of the constructor? Or is there something we could do so that we only instantiate it once?